fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/Statistics.java

```diff
@@ -33,11 +35,13 @@ public class Statistics {
     // Table row count in FE depends on BE reporting，but FE may not get report from BE which just started，
     // this causes the table row count stored in FE to be inaccurate.
     private final boolean tableRowCountMayInaccurate;
+    private final Collection<ColumnRefOperator> shadowColumns;
```

Comment:
What are shadow columns? columns that are real needed after column prune?

Reply:
we add a projection to keep the output columns of rewritten plan with original plan, which will add cost for rewrite plan, so I exclude it from cost calculation.